spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LSTMPeephole.scala

```diff
@@ -67,13 +67,16 @@ class LSTMPeephole[T : ClassTag] (
   var cellLayer: Sequential[T] = _
   override var cell: AbstractModule[Activity, Activity, T] = buildLSTM()
 
-  def buildGate(): Sequential[T] = {
+  override def preTopology: AbstractModule[Activity, Activity, T] =
+    Sequential()
+    .add(Dropout(p))
+    .add(TimeDistributed(Linear(inputSize, hiddenSize * 4, wRegularizer = wRegularizer,
+      bRegularizer = bRegularizer)))
+
+  def buildGate(dimension: Int, offset: Int, length: Int): Sequential[T] = {
     val gate = Sequential()
 
-    val i2g = Sequential()
-      .add(Dropout(p))
-      .add(Linear(inputSize, hiddenSize, wRegularizer = wRegularizer,
-        bRegularizer = bRegularizer))
+    val i2g = Narrow(dimension, offset, length)
```

Comment:
We use Narrow to select the corresponding input instead of calculating the Linear layer.
For example, the input size is [3, 12]. Then we need to select the sub-tensor from the second dimension: [3, 4], [3, 4], [3, 4].
Do you have any suggestions?